File: pkgs/native_toolchain_c/test/cbuilder/testfiles/add_objective_c/src/add.m
// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

#import <Foundation/Foundation.h>

#if DEBUG
#import <stdio.h>
#endif

NSInteger add(NSInteger a, NSInteger b) {
#ifdef DEBUG
  printf("Adding %ld and %ld.\n", (long)a, (long)b);
#endif
  // Use NSNumber from Foundation to test the frameworks flags.
  return [[NSNumber numberWithInteger:(a + b)] longValue];
}

File: pkgs/native_toolchain_c/test/cbuilder/testfiles/add/src/add.c
// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

#include <stdint.h>

#ifdef DEBUG
#include <stdio.h>
#endif

#if _WIN32
#define FFI_EXPORT __declspec(dllexport)
#else
#define FFI_EXPORT
#endif

FFI_EXPORT int32_t add(int32_t a, int32_t b) {
#ifdef DEBUG
  printf("Adding %i and %i.\n", a, b);
#endif
  return a + b;
}
